### PR TITLE
Workaround pow implentation bugs when one of the arguments is NaN

### PR DESCRIPTION
**Pull request by stinos**
This PR is just to get the fix publicly visible, as far as I'm concerned the 2 commits can be merged.

Note: the code is written out in full in objfloat.c instead of reusing the line with `lhs_val = ...` because otherwise code formatting trips over it and messes up indentation of the closing brace.

### CODE REVIEW

File: py/mpconfig.h [hunk 1]
```diff
@@ -1160,6 +1160,11 @@ typedef double mp_float_t;
 #define MICROPY_PY_MATH_MODF_FIX_NEGZERO (0)
 #endif
 
+// Whether to provide fix for pow(1, NaN), which should be 1 not NaN.
```
Comment (dpgeorge) [inline py/mpconfig.h:1163]: Perhaps the comment here should match that in modmath, eg "Whether to provide a fix for pow(1, NaN) and pow(NaN, 0), which should both be 1"

File: py/objfloat.c [hunk 2]
```diff
@@ -293,14 +293,22 @@ mp_obj_t mp_obj_float_binary_op(mp_binary_op_t op, mp_float_t lhs_val, mp_obj_t
...
+            if (lhs_val < 0 && rhs_val != MICROPY_FLOAT_C_FUN(floor)(rhs_val) && !isnan(rhs_val)) {
                 #if MICROPY_PY_BUILTINS_COMPLEX
                 return mp_obj_complex_binary_op(MP_BINARY_OP_POWER, lhs_val, 0, rhs_in);
                 #else
                 mp_raise_ValueError(MP_ERROR_TEXT("complex values not supported"));
                 #endif
             }
+            #if MICROPY_PY_MATH_POW_FIX_NAN // Also see modmath.c.
+            if (lhs_val == MICROPY_FLOAT_CONST(1.0) || rhs_val == MICROPY_FLOAT_CONST(0.0)) {
+                lhs_val = MICROPY_FLOAT_CONST(1.0);
```
Comment (dpgeorge) [inline py/objfloat.c:305]: If you add a `break` after this line then this fix could be self contained, ie no need to duplicate the `lhs_val = MICROPY_FLOAT_C_FUN(pow)(lhs_val, rhs_val);` line below.

**Comment by dpgeorge** [comment]
Thanks for this, it's nice and clean.

> as far as I'm concerned the 2 commits can be merged

IMO they should stay separate: the first one fixes a bug in the existing code, whereas the second one is a workaround for a bug in a C library.

File: py/mpconfig.h (hunk 1 above)
Comment (stinos) [reply py/mpconfig.h:1163]: Spot on, I wrote this before I found the other bug

File: py/objfloat.c (hunk 2 above)
Comment (stinos) [reply py/objfloat.c:305]: That's better indeed

**Comment by stinos** [comment]
Ok makes sense.

**Comment by dpgeorge** [comment]
Thanks for updating, merged in 8d5a40c86e384bf3cddb2f687374e0bb1ae6df7d and 2e54d9d146b34d7ad00e4394c9767f4319244cdf

**Pull request closed without merging**